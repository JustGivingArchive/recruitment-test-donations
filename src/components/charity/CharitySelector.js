import React from 'react';
import IndividualCharity from './IndividualCharity.js';

// donations are served from the API as an array, so we can map over them
// and create a donation component for each

const CharitySelector = (props) => {
  const charitys = [{
    name: 'British Heart Foundation',
    id: '183092'
  },
  {
    name: 'Cancer Research UK',
    id: '2357'
  },
  {
    name: 'Macmillan Cancer Support',
    id: '2116'
  },
  {
    name: 'Oxfam',
    id: '13441'
  },
  {
    name: 'National Trust',
    id: '183560'
  },
  {
    name: 'Save the Children',
    id: '18570'
  }
  ];
  return (
    <div className="content_container">
      {charitys.map((charity) => <IndividualCharity select={props.select} {...charity} />)}
    </div>
  );
};

export default CharitySelector;
