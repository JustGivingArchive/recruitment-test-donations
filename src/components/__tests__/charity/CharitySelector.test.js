import React from 'react';
import renderer from 'react-test-renderer';
import CharitySelector from '../../charity/CharitySelector.js';

describe('Test charity info componenet', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <CharitySelector />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
