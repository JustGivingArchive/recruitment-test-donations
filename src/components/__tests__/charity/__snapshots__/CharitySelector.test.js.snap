exports[`Test charity info componenet renders correctly 1`] = `
<div
  className="content_container">
  <div
    className="panel hoverable"
    onClick={[Function]}>
    <p
      className="paneltextlarge">
      British Heart Foundation
    </p>
  </div>
  <div
    className="panel hoverable"
    onClick={[Function]}>
    <p
      className="paneltextlarge">
      Cancer Research UK
    </p>
  </div>
  <div
    className="panel hoverable"
    onClick={[Function]}>
    <p
      className="paneltextlarge">
      Macmillan Cancer Support
    </p>
  </div>
  <div
    className="panel hoverable"
    onClick={[Function]}>
    <p
      className="paneltextlarge">
      Oxfam
    </p>
  </div>
  <div
    className="panel hoverable"
    onClick={[Function]}>
    <p
      className="paneltextlarge">
      National Trust
    </p>
  </div>
  <div
    className="panel hoverable"
    onClick={[Function]}>
    <p
      className="paneltextlarge">
      Save the Children
    </p>
  </div>
</div>
`;
